File: UCF HSPT Documents/2013/Solutions/oldman.c

// 27th Annual UCF High School Programming Tournament
//
// Problem Name: Old Man of the Mountain (oldman)
//
// Author:        Glenn Martin & Matt Fontaine
// Judge Data:    Glenn Martin
// C Solution:    Glenn Martin
// Java Solution: Glenn Martin
// Verifier:      Glenn Martin

#include <stdio.h>


int main()
{
   FILE *   infile;
   int      numSites;
   int      i;
   int      m1, m2, m3;

   // Open the input file
   infile = fopen("oldman.in", "r");

   // Get number of sites to test
   fscanf(infile, "%d", &numSites);

   // Loop over the sites
   for (i=0; i < numSites; i++)
   {
      // Get the three measurements
      fscanf(infile, "%d %d %d", &m1, &m2, &m3);

      // Test to see if they match
      printf("Set #%d: ", i+1);
      if ( (m1 == m2) && (m1 == m3) && (m2 == m3) )
         printf("Sorry, too flat.\n");
      else
         printf("Shows potential!\n");
   }

   // Done
   fclose(infile);
   return 0;
}

File: UCF HSPT Team/src/y2013/oldman.java

// 27th Annual UCF High School Programming Tournament
//
// Problem Name: Old Man of the Mountain (oldman)
//
// Author:        Glenn Martin & Matt Fontaine
// Judge Data:    Glenn Martin
// C Solution:    Glenn Martin
// Java Solution: Glenn Martin
// Verifier:      Glenn Martin

import java.io.File;
import java.io.IOException;
import java.util.Scanner;


public class oldman 
{
   public static void main(String[] args) throws IOException 
   {
      // Open file for reading
      Scanner in = new Scanner(new File("oldman.in"));

      // Get in the number of sites
      int numSites = in.nextInt();

      // Loop over the sites
      for (int i=0; i < numSites; i++)
      {
         // Get the three measurements
         int m1 = in.nextInt();
         int m2 = in.nextInt();
         int m3 = in.nextInt();

         // Test them
         System.out.printf("Set #%d: ", i+1);
         if ( (m1 == m2) && (m1 == m3) && (m2 == m3) )
            System.out.printf("Sorry, too flat.%n");
         else
            System.out.printf("Shows potential!%n");
      }

      // Done
      in.close();
   }
}
